README.md

```diff
@@ -36,16 +36,49 @@ you must call it multiple times in order to build multiple architectures.
 
 ### Action Inputs Explained
 
-- **version** is the product version we are building a docker image for.
-- **target** is the name of the "stage" or "target" in the Dockerfile to build.
-- **arch** is the architecture we're building for.
-- **tags** is a newline-separated list of the "production tags" for this image.
+- **`version`** is the product version we are building a docker image for.
+- **`target`** is the name of the "stage" or "target" in the Dockerfile to build.
+- **`arch`** is the architecture we're building for.
+- **`tags`** is a newline-separated list of the "production tags" for this image.
   **Note that you must define the same tags for each architecture you're building,
   so the tag should never reference the architecture. See note below.**
-- **dev_tags** is similar to **tags** except these tags are not intended for
+- **`dev_tags`** is similar to **tags** except these tags are not intended for
   production/final releases; **dev_tags** are typically published much more
   frequently than production tags, and are used for early access to the latest code.
 
+#### Note on `target`
+
+The `target` input is provided so that you can define multiple images inside the
+same `Dockerfile`. Here is an example Dockerfile that defines two independent targets,
+`target1` and `target2`:
+
+```Dockerfile
+FROM alpine:latest AS target1
+# Lines omitted.
+
+FROM ubuntu:latest AS target2
+# Lines omitted.
+```
+
+Sometimes you'll only want one release target (which must `COPY` a local product
+binary inside it), and one "local dev" target which actually performs the build from
+source for local development purposes. In this case you would only reference the
+release target in the action configuration.
+
+Some products need to produce multiple release Docker images. In this case, you
+can reference each different image by its target name, in a separate call to this
+action. You must be careful to ensure that the tags are different for images built
+from different targets.
```

Comment:
We should add a check that the tag names don't clash. It's bound to come up eventually and be very confusing... @alvin-huang FYI

Reply:
Tracked here: https://hashicorp.atlassian.net/browse/RELTOOL-105